source/PlayerInfo.cpp

```diff
@@ -4005,6 +4039,31 @@ void PlayerInfo::CreateMissions()
 
 
 
+// New random stock is generated each time you land on a planet.
+void PlayerInfo::CreateRandomStock()
+{
+	int day = GetDate().DaysSinceEpoch();
+	for(const auto &rStock : planet->OutfitRandomStock())
+		for(const auto &stockItem : *rStock)
+			if(!planet->Outfitter().Has(stockItem.item) && Random::Int(100) < stockItem.probability)
```

Comment:
If you agree with my [comment](https://github.com/endless-sky/endless-sky/pull/10265#discussion_r1686327438) on the original PR
```suggestion
			if(Random::Int(100) < stockItem.probability)
```